From: =?utf-8?q?Rapha=C3=ABl_Hertzog?= <raphael@offensive-security.com>
Date: Tue, 25 Jun 2019 11:19:30 +0200
Subject: Print a deprecation notice when luksAddNuke is called

The notice invites the user to run "dpkg-reconfigure
cryptsetup-nuke-password" after having installed that package.
---
 src/cryptsetup.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/cryptsetup.c b/src/cryptsetup.c
index 601a967..99a98ac 100644
--- a/src/cryptsetup.c
+++ b/src/cryptsetup.c
@@ -1838,6 +1838,15 @@ out:
 	return r;
 }
 
+static int action_luksAddNuke(void)
+{
+	fprintf(stderr, "ERROR: 'cryptsetup luksAddNuke' is deprecated.\n");
+	fprintf(stderr, "Instead use the feature provided by the cryptsetup-nuke-password package:\n");
+	fprintf(stderr, " sudo apt install cryptsetup-nuke-password\n");
+	fprintf(stderr, " sudo dpkg-reconfigure cryptsetup-nuke-password\n");
+	return -1;
+}
+
 static int action_luksChangeKey(void)
 {
 	const char *opt_new_key_file = (action_argc > 1 ? action_argv[1] : NULL);
@@ -3294,6 +3303,7 @@ static struct action_type {
 	{ "config",       action_luksConfig,   1, 1, N_("<device>"), N_("set permanent configuration options for LUKS2") },
 	{ "luksFormat",   action_luksFormat,   1, 1, N_("<device> [<new key file>]"), N_("formats a LUKS device") },
 	{ "luksAddKey",   action_luksAddKey,   1, 1, N_("<device> [<new key file>]"), N_("add key to LUKS device") },
+	{ "luksAddNuke",  action_luksAddNuke,  0, 0, N_("<device> [<new key file>]"), N_("print deprecation warning about nuke feature") },
 	{ "luksRemoveKey",action_luksRemoveKey,1, 1, N_("<device> [<key file>]"), N_("removes supplied key or key file from LUKS device") },
 	{ "luksChangeKey",action_luksChangeKey,1, 1, N_("<device> [<key file>]"), N_("changes supplied key or key file of LUKS device") },
 	{ "luksConvertKey",action_luksConvertKey,1, 1, N_("<device> [<key file>]"), N_("converts a key to new pbkdf parameters") },
